design-documents/web-api-controllers.md

```diff
@@ -0,0 +1,374 @@
+# Controllers for web APIs, user input processing on top of service contracts execution
+## RESTful and SOAP API
+#### Current situation
+Service contracts, their arguments and return types are supposed to be used directly
+as web API endpoints. The problem is service contracts are also used as API
+and are supposed to be a part of business layer. By using them for presentation developers may encounter various
+problems like having to put additional logic related to specifically processing web API requests (fields validation
```

Comment:
not sure what do you mean?

Reply:
`fields validation and are supposed to be part of business layer` - this is controversial from my point of view since there is no technical vision for validation